### PR TITLE
docs: add seealso section to gzip sandbox

### LINKED ISSUE #16748: Add "see also" section to gzip sandbox
### description

it would be helpful to add a see also section to the gzip sandbox pointing to:

- api docs
- any reference documentation

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

Commit Message:

Add seealso section to gzip sandbox

Additional Description:
Risk Level: LOW
Testing:
Docs Changes: add seealso section.
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Fix #16748
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
looks good @daixiang0 - could we also add a link to the gzip api docs please

File: docs/root/start/sandboxes/gzip.rst [hunk 1]
```diff
@@ -85,8 +85,10 @@ Now, use ``curl`` to make a request for the compressed statistics:
     content-encoding: gzip
 
 .. seealso::
+   :ref:`Gzip Api <envoy_v3_api_field_extensions.filters.http.gzip.v3.Gzip.compressor>`
+      Learn more abort Envoy’s gzip Apis.
 
-   :ref:`gzip filter <config_http_filters_gzip>`.
+   :ref:`Gzip configuration <config_http_filters_gzip>`.
       Learn more about configuring Envoy's gzip filter.
```
Comment (phlax) [inline docs/root/start/sandboxes/gzip.rst:92]: i would make this something like "Reference documentation for Envoy's gzip compression"

File: docs/root/start/sandboxes/gzip.rst [hunk 2]
```diff
@@ -85,8 +85,10 @@ Now, use ``curl`` to make a request for the compressed statistics:
     content-encoding: gzip
 
 .. seealso::
+   :ref:`Gzip Api <envoy_v3_api_field_extensions.filters.http.gzip.v3.Gzip.compressor>`
+      Learn more abort Envoy’s gzip Apis.
```
Comment (phlax) [inline docs/root/start/sandboxes/gzip.rst:89]: and this one i think something like "API and configuration reference for Envoy's gzip compression"

File: docs/root/start/sandboxes/gzip.rst [hunk 3]
```diff
@@ -83,3 +83,13 @@ Now, use ``curl`` to make a request for the compressed statistics:
 
     $ curl -si -H "Accept-Encoding: gzip" localhost:9902/stats/prometheus | grep "content-encoding"
     content-encoding: gzip
+
+.. seealso::
+   :ref:`Gzip Api <envoy_v3_api_field_extensions.filters.http.gzip.v3.Gzip.compressor>`
+      API and configuration reference for Envoy's gzip compression.
+
+   :ref:`Gzip configuration <config_http_filters_gzip>`.
+      Reference documentation for Envoy's gzip compression.
```
Comment (rojkov) [inline docs/root/start/sandboxes/gzip.rst:92]: These two are deprecated. Better reference the generic compressor filter configured with gzip as a compression library.

Comment (daixiang0) [reply docs/root/start/sandboxes/gzip.rst]: I have updated it.

**Comment by phlax** [comment]
rendered docs are here https://storage.googleapis.com/envoy-pr/16796/docs/index.html

@daixiang0 could you fix format please

**Comment by phlax** [comment]
probably you want to adjust your editor so these dont happen

```
ERRORS found
1 files with missing newline
0 files with mixed tabs and spaces
0 files with trailing whitespace
```

File: docs/root/start/sandboxes/gzip.rst [hunk 4]
```diff
@@ -83,3 +83,13 @@ Now, use ``curl`` to make a request for the compressed statistics:
 
     $ curl -si -H "Accept-Encoding: gzip" localhost:9902/stats/prometheus | grep "content-encoding"
     content-encoding: gzip
+
+.. seealso::
+   :ref:`Gzip Api <envoy_v3_api_msg_extensions.compression.gzip.compressor.v3.Gzip>`
```
Comment (phlax) [inline docs/root/start/sandboxes/gzip.rst:88]: could you change `Api` -> `API`

File: docs/root/start/sandboxes/gzip.rst [hunk 5]
```diff
@@ -83,3 +83,13 @@ Now, use ``curl`` to make a request for the compressed statistics:
...
+      API and configuration reference for Envoy's gzip compression.
+
+   :ref:`Gzip configuration <config_http_filters_compressor>`.
```
Comment (phlax) [inline docs/root/start/sandboxes/gzip.rst:91]: could you change `Gzip` -> `Compression`

File: docs/root/start/sandboxes/gzip.rst [hunk 6]
```diff
@@ -83,3 +83,13 @@ Now, use ``curl`` to make a request for the compressed statistics:
...
+      Reference documentation for Envoy's gzip compression.
```
Comment (phlax) [inline docs/root/start/sandboxes/gzip.rst:92]: could you change `gzip compression` -> `compressor filter`

**Review by phlax — CHANGES_REQUESTED** [review]
almost there @daixiang0 a few nits below...

**Review by phlax — APPROVED** [review]
lgtm, thanks @daixiang0

**Pull request merged**